{% extends "root.html" %}

{%block title%}Documentation{%endblock%}
{%block savePageIcon%}{%endblock%}

{%block main%}

{%include "snippets/messages.html"%}

<div class='container-fluid'>
	<div class='row' id='top'>
		<div class='col-md-12 white'>
			<div id='about' class="panel panel-default black">
				<div class="panel-heading">
    					<h3 class="panel-title">About</h3>
  				</div>
  				<div class="panel-body">
					<p class="text-justify">OREAN stands for <b><u>O</u></b>mics <b><u>RE</u></b>search <b><u>AN</u></b>alytics and serves as a resource for analysis, visualization, and sharing of multi-omic microbiome datasets.</p>
  				</div>
			</div>

                        <div id='terminology' class="panel panel-default black">
                                <div class="panel-heading">
                                        <h3 class="panel-title">Terminology</h3>
                                </div>
                                <div class="panel-body">
                                        <blockquote id='terminology-generic'><h5>Generic</h5></blockquote>

					<dl class="dl-horizontal">
						<dt>Query</dt>
						<dd>A set of samples from a single project. Queries are computed at the time of request and samples matching the criteria for the query are stored.</dd>
						<dd>The criteria mandated for a particular query is not stored by the application. Thus is is recommended to adopt meaningful names and to provide detailed descriptions of how each query was prepared.</dd>

						<dt>Attribute Data</dt>
						<dd>Stores in the application any information available for a particular sample. The field can support numeric and non-numeric data as well as dates.</dd>

						<dt>Analysis Data</dt>
						<dd>Stores results obtained by running various analysis protocols. Can support multiple different analysis techniques for a particular dataset.</dd>
						<dd>Optional inclusion of NCBI tax IDs for heirarchical data is supported.</dd>
					</dl>
                                </div>
                        </div>

                        <div id='uploads' class="panel panel-default black">
                                <div class="panel-heading">
                                        <h3 class="panel-title">Uploads</h3>
                                </div>
                                <div class="panel-body">

					<p class='text-justify'>The content below describes the file formats required to add data into the application. Example input data as well as use cases have been provided for a 16S taxonomic classification analysis. However, the format has been designed in a flexible way in an effort to accomodate many different types of multi-omic datasets.</p>

					<blockquote id='uploads-attributedata'><h5>Attribute Data</h5></blockquote>

					<dl class="dl-horizontal">
                                                <dt>Sample</dt>
                                                <dd>The name of the sample.</dd>
                                                <dd>This file uses a "columnar" style. This means that each sample name should appear on multiple lines, one for each field.</dd>

                                                <dt>Category</dt>
                                                <dd>A classification for the attribute. Can be any string.</dd>

                                                <dt>Field</dt>
                                                <dd>The name of the field. Can be any string</dd>

                                                <dt>Value</dt>
                                                <dd>The value for the field. Values can be numbers, characters, and dates. If a particular field name has different data types for different samples, the field will be considered character data.</dd>
                                        </dl>
				<div class='row'>
				<div class='col-md-6'>	<pre>
<strong>Sample   Category   Field             Value</strong>
sample1  metadata   race              African American
sample1  metadata   gender            Male
sample1  metadata   age               [60-70)
sample1  metadata   time_in_hospital  [1.5-3.5)
sample1  diagnosis  diag_1            [0.0-1.5)
sample1  diagnosis  diag_2            157
sample1  diagnosis  number_diagnoses  197
sample2  metadata   race              Caucasian
sample2  metadata   gender            Female
sample2  metadata   age               [40-50)
sample2  metadata   time_in_hospital  [3.5-14.0]
sample2  diagnosis  diag_1            [0.0-1.5)
sample2  giagnosis  diag_2            428
sample2  diagnosis  number_diagnoses  250.6
</pre></div>

                               <div class='col-md-6'>
					<p>The columnar format allows for customized categorization of metadata field groups, but often this data is stored in a matrix style table. For convenience, both formats are supported. If a matrix style table is provided the application will assign all the variables to the generic category "metadata". In the matrix format, the first column must be the sample name followed by the field values for that sample. A single header line defines the name of each field.</p>
<pre><strong>sample   race              gender  age      time_in_hospital  diag_1     diag_2  number_diagnoses</strong>
sample1  African American  Male    [60-70)  [1.5-3.5)         [0.0-1.5)  157     197
sample2  Caucasian         Female  [40-50)  [3.5-14.0]        [0.0-1.5)  428     250.6
</pre></div>
</div>
					<hr>
					<blockquote id='uploads-analysisdata'><h5>Analysis Data</h5></blockquote>
					<dl class="dl-horizontal">
                                               <dt>Sample</dt>
                                                <dd>The name of the sample.</dd>
                                                <dd>This file uses a "columnar" style. This means that each sample name should appear on multiple lines. In "16S" data the sample appears once for each classification.</dd>

                                                <dt>Dataset</dt>
                                                <dd>A broad term indicating the type of the data. Can be any string. Something like "16S" would be appropriate.</dd>

                                                <dt>Method</dt>
                                                <dd>The specific technique used for the analysis. This may be the name of a specific software or analysis technique. For "16S" data the method may be the name of the classification software that was used.</dd>

                                                <dt>Category</dt>
                                                <dd>This field stores the level information in heirarchical datasets. For 16S data this field should represent the taxa level of the classification.</dd>
                                                <dd>If the data does not have multiple levels the same string may be provided for all rows.</dd>

                                                <dt>Taxa</dt>
                                                <dd>This field stores the name of organism into which the sample has been assigned.</dd>
                                                
                                                <dt>TaxId</dt>
                                                <dd>The taxonomy ID for the classification, if applicable. If this information does not apply or is not available, this field should be populated with "NA".</dd>

                                                <dt>Count</dt>
                                                <dd>This field must be a number. It represents the raw result of the analysis for the particular sample prior to any sort of scaling or normalization.</dd>
						<dd>For a 16S classification profile, this would represent a count of the number of reads classified to the particular taxonomy</dd>

                                                <dt>Normalized</dt>
                                                <dd>This is the scaled or normalized result for the sample.</dd>
						<dd>For a 16S classification profile, this would represent the percentage of reads that classified to the particular taxonomy</dd>
                                                
                                                <dt>Avg_Confidence</dt>
                                                <dd>If the analysis provides a statistical confidence value for the results that are generated, that value should be placed here.</dd>
						<dd>This field must be a number. If no confidence score is available then any generic number value should be placed here for all samples.</dd>
                                        </dl>
					<div class='row'>
					<div class='col-md-6'>
					<pre>
<strong>Sample   Dataset   Method   Category  Taxa              TaxId  Count  Normalized  Avg_Confidence</strong>
sample1  Data-16s  RDP-0-8  rootrank  Root              0      3989   100         1
sample1  Data-16s  RDP-0-8  domain    Bacteria          1      3989   100         1
sample1  Data-16s  RDP-0-8  phylum    Actinobacteria    2      221    5.54        1
sample1  Data-16s  RDP-0-8  class     Actinobacteria    3      221    5.54        1
sample1  Data-16s  RDP-0-8  subclass  Actinobacteridae  16     147    3.69        0.99
sample1  Data-16s  RDP-0-8  order     Actinomycetales   17     147    3.69        0.99
sample1  Data-16s  RDP-0-8  suborder  Actinomycineae    18     94     2.36        0.98
sample1  Data-16s  RDP-0-8  family    Actinomycetaceae  19     94     2.36        0.98
sample1  Data-16s  RDP-0-8  genus     Actinomyces       21     92     2.31        0.98
sample2  Data-16s  RDP-0-8  rootrank  Root              0      13173  100         1
sample2  Data-16s  RDP-0-8  domain    Bacteria          1      13138  99.73       1
sample2  Data-16s  RDP-0-8  phylum    Actinobacteria    2      3861   29.31       0.99
sample2  Data-16s  RDP-0-8  class     Actinobacteria    3      3861   29.31       0.99
sample2  Data-16s  RDP-0-8  subclass  Actinobacteridae  16     3834   29.1        0.99
sample2  Data-16s  RDP-0-8  order     Actinomycetales   17     3826   29.04       0.99
sample2  Data-16s  RDP-0-8  suborder  Actinomycineae    18     359    2.73        0.98
sample2  Data-16s  RDP-0-8  family    Actinomycetaceae  19     359    2.73        0.98
sample2  Data-16s  RDP-0-8  genus     Actinomyces       21     354    2.69        0.98
</pre></div>
						
					<div class='col-md-6'>
						<p>Like the metadata, both columnar and matrix style formats are supported for the analysis data. The columnar format allows multiple dataset, method, and categories to be provided in a single file. When uploading a matrix formatted file, this information will be entered into the upload form and is therefore limited to a single value for the entire file. Taxonomy is not supported for matrix files nor is confidence values. The values provided should be the normalized values if a matrix file is provided.</p>
						<pre><strong>sample   Firmicutes  Actinobacteria  Bacteroidetes  Tenericutes</strong>
sample1  40.50       15.50           35.20          8.80
sample2  22.70       10.45           14.7           52.15</pre>
					</div>
                                </div>
                        </div>
                        </div>

                        <div id='contact' class="panel panel-default black">
                                <div class="panel-heading">
                                        <h3 class="panel-title">Contact</h3>
                                </div>
                                <div class="panel-body">
					<p>
					Nihar Sheth
					<br>Technical Director, BCCL
					<br>P: (804) 827-0951
					<br>Email: <a href='mailto:nsheth@vcu.edu'>nsheth@vcu.edu</a>
					</p>
                                </div>
                        </div>

                        <div id='citation' class="panel panel-default black">
                                <div class="panel-heading">
                                        <h3 class="panel-title">Citation</h3>
                                </div>
                                <div class="panel-body">
					Github Repository: <a href="https://github.com/nisheth/OREAN">https://github.com/nisheth/OREAN</a>
                                </div>
                        </div>

		</div>
	</div>
</div>
{%endblock%}

